### PR TITLE
[MAEB] Evaluation Fixes

### PR DESCRIPTION
**Pull request by AdnanElAssadi56**
I’ve added some model handling to make models more efficient and safety checks so evaluations can continue without getting stuck. Not all of these checks may be strictly necessary, but I added them incrementally as issues came up, along with fallbacks where needed. @gowitheflow-1998 should be able to checkpoint from here and complete the remaining evaluations without major issues.

Some datasets had typos in audio label names; for others, I uploaded subsampled versions and evaluated on those. I also added some data handling to deal with bugs in some dataset transform functions. There are probably places where this could be cleaned up further, but I think this is sufficient to complete the evaluations.

@Samoed feel free to make changes directly if you think anything should be improved, but I believe what we have now is fine. Some errors are triggered only by specific audio–model combinations while others are more global.

### CODE REVIEW

File: mteb/models/model_implementations/cnn14_model.py [hunk 1]
```diff
@@ -12,6 +12,18 @@
 from mteb.types import Array, BatchedInput, PromptType
 from mteb.types._encoder_io import AudioInput
 
+# Monkey-patch torchaudio.list_audio_backends if missing
```
Comment (Samoed) [inline mteb/models/model_implementations/cnn14_model.py:15]: When this is happening?

File: mteb/models/model_implementations/ast_model.py [hunk 2]
```diff
@@ -66,6 +66,13 @@ def get_audio_embeddings(
                         orig_freq=sr, new_freq=self.sampling_rate
                     )
                     array = resampler(array)
+                
+                # Ensure minimum length for AST feature extractor (window size is 400)
```
Comment (Samoed) [inline mteb/models/model_implementations/ast_model.py:70]: Can this be done through processor?

File: mteb/models/model_implementations/mctct_model.py [hunk 3]
```diff
@@ -178,6 +178,11 @@ def get_audio_embeddings(
                 valid_tokens = hidden_attention_mask.sum(dim=1)
                 embeddings = masked_embeddings.sum(dim=1) / valid_tokens.clamp(min=1e-9)
 
+                # Replace any NaNs with zeros to prevent crashes in downstream classifiers
```
Comment (Samoed) [inline mteb/models/model_implementations/mctct_model.py:181]: I don't think we want to fill NaNs. Probably it's better warn users of incompatible torch version

File: mteb/models/model_implementations/seamlessm4t_models.py [hunk 4]
```diff
@@ -68,24 +68,26 @@ def get_audio_embeddings(
...
+            )
+
+            input_features = features.input_features.to(self.device)
+            attention_mask = (
+                features.attention_mask.to(self.device)
+                if hasattr(features, "attention_mask")
+                else None
+            )
+
+            with torch.no_grad():
```
Comment (Samoed) [inline mteb/models/model_implementations/seamlessm4t_models.py:90]: Model is already in eval mode

**Review by Samoed — COMMENTED** [review]
I briefly saw sime model implementations. Will look deeper later

**Comment by Samoed** [comment]
It would be better if you created separate PRs with fixes for each model and task.

File: mteb/models/model_implementations/cnn14_model.py (hunk 1 above)
Comment (AdnanElAssadi) [reply mteb/models/model_implementations/cnn14_model.py:15]: It caused an error for me when evaluating cnn14, probably related to the `torchaudio` version. I think `speechbrain` triggers old api and tries to use list_audio_backends.

File: mteb/models/model_implementations/ast_model.py (hunk 2 above)
Comment (AdnanElAssadi) [reply mteb/models/model_implementations/ast_model.py:70]: Not quite sure

File: mteb/models/model_implementations/cnn14_model.py (hunk 1 above)
Comment (Samoed) [reply mteb/models/model_implementations/cnn14_model.py:15]: I think we need to pin version for this model then

File: mteb/models/model_implementations/ast_model.py (hunk 2 above)
Comment (Samoed) [reply mteb/models/model_implementations/ast_model.py:70]: Yeah, seems that it supports only max sampling rate https://github.com/huggingface/transformers/blob/a7f29523361b2cc12e51c1f5133d95f122f6f45c/src/transformers/models/audio_spectrogram_transformer/feature_extraction_audio_spectrogram_transformer.py#L50-L62

File: mteb/models/model_implementations/encodec_model.py [hunk 5]
```diff
@@ -76,22 +78,46 @@ def get_audio_embeddings(
                     array = resampler(array)
 
                 array = array.squeeze()
+                
+                # Handle edge case where squeeze results in 0-dim tensor
+                if array.dim() == 0:
+                    array = array.unsqueeze(0)
+                
+                # Ensure we have at least some samples
+                if len(array) == 0:
```
Comment (Samoed) [inline mteb/models/model_implementations/encodec_model.py:87]: Do we have such tasks? I don't think that this should happen

File: mteb/models/model_implementations/encodec_model.py [hunk 6]
```diff
@@ -76,22 +78,46 @@ def get_audio_embeddings(
...
+                if len(array) < min_samples:
+                    padding = torch.zeros(min_samples - len(array))
+                    array = torch.cat([array, padding])
+                
+                # Pad or truncate to fixed length for consistent batching
+                if len(array) > max_samples:
+                    array = array[:max_samples]
+                elif len(array) < max_samples:
+                    padding = torch.zeros(max_samples - len(array))
+                    array = torch.cat([array, padding])
```
Comment (Samoed) [inline mteb/models/model_implementations/encodec_model.py:104]: You can process max size and padding with processor https://github.com/huggingface/transformers/blob/a7f29523361b2cc12e51c1f5133d95f122f6f45c/src/transformers/models/encodec/feature_extraction_encodec.py#L87-L91

File: mteb/models/model_implementations/encodec_model.py [hunk 7]
```diff
@@ -76,22 +78,46 @@ def get_audio_embeddings(
...
+                
                 audio_arrays.append(array.numpy())
 
             with torch.no_grad():
-                # Process audio through EnCodec's processor
-                max_samples = int(self.max_audio_length_seconds * self.sampling_rate)
-
-                feature_inputs = self.processor(
```
Comment (Samoed) [inline mteb/models/model_implementations/encodec_model.py:85]: Why do you remove processor?

File: mteb/models/model_implementations/muq_mulan_model.py [hunk 8]
```diff
@@ -98,13 +98,23 @@ def get_audio_embeddings(
...
-        text_embeds = self.model(texts=texts)
+        all_embeddings = []
+
+        for batch in tqdm(
+            inputs,
+            disable=not show_progress_bar,
+            desc="Processing text batches",
+        ):
+            texts = batch["text"]
+            with torch.no_grad():
```
Comment (Samoed) [inline mteb/models/model_implementations/muq_mulan_model.py:113]: It's already in eval mode
```suggestion
```

File: mteb/models/model_implementations/wav2clip_model.py [hunk 9]
```diff
@@ -34,12 +34,11 @@ def __init__(
         self.sampling_rate = 16_000
         self.max_audio_length_s = max_audio_length_s
 
-        # text side (CLIP)
-        self.clip = CLIPModel.from_pretrained(model_name, revision=revision).to(device)
+        # text side (CLIP) - we use the standard OpenAI CLIP model as mentioned in paper
+        clip_model_name = "openai/clip-vit-base-patch32"
+        self.clip = CLIPModel.from_pretrained(clip_model_name).to(device)
```
Comment (Samoed) [inline mteb/models/model_implementations/wav2clip_model.py:39]: Can you add revision?

File: mteb/tasks/audio/any_2_any_retrieval/common_voice.py
Comment (Samoed) [inline mteb/tasks/audio/any_2_any_retrieval/common_voice.py:1]: It this same as in https://github.com/embeddings-benchmark/mteb/pull/3709?

File: mteb/tasks/audio/audio_multilabel_classification/eng/fsd50_hf.py [hunk 10]
```diff
@@ -8,12 +8,12 @@
 class FSD50HFMultilingualClassification(AbsTaskMultilabelClassification):
     metadata = TaskMetadata(
         name="FSD50K",
-        description="Multilabel Audio Classification.",
-        reference="https://huggingface.co/datasets/Chand0320/fsd50k_hf",
+        description="Multilabel Audio Classification on a subsampled version of FSD50K.",
```
Comment (Samoed) [inline mteb/tasks/audio/audio_multilabel_classification/eng/fsd50_hf.py:11]: Can we keep both versions? Can you move to another PR for discussion?

**Comment by Samoed** [comment]
I'll clsoe this because thit pr was splitted
